tools/src/main/scala/scala/scalanative/build/Build.scala

```diff
@@ -136,4 +184,18 @@ object Build {
     }
   }
 
+  private def getLastModifiedTimeMillis(path: Path): Long =
+    if (Files.exists(path)) Files.getLastModifiedTime(path).toMillis()
+    else 0L
+
+  /** Get the newest file's last modified time in millis under the given path.
+   */
+  private def getNewestMtime(path: Path): Optional[FileTime] =
+    if (Files.exists(path))
+      Files
+        .walk(path, FileVisitOption.FOLLOW_LINKS)
+        .map[FileTime](Files.getLastModifiedTime(_))
+        .max(_.compareTo(_))
```

Comment:
I forget we can use prediefined comparator here: java.util.Comparator.naturalOrder

Reply:
Remained same as `_max(_.compareto(_))` since `java.util.Comparator.naturalOrder` doesn't work for `FileTime`, or do you mean we can make this function return `Long` instead of `FileTime` (I think we can because we just calculate hashCode from `FileTime` instance).